src/nvim/os/env.c

```diff
@@ -480,9 +478,9 @@ char_u *vim_getenv(char_u *name, bool *mustfree)
 
       // check that the result is a directory name
       assert(pend >= p);
-      p = vim_strnsave(p, (size_t)(pend - p));
+      p = (char *)vim_strnsave((char_u *)p, (size_t)(pend - p));
```

Comment:
This PR is just for replacing char_u -- but I have one that is separate to more fully refactor the vim_getenv() function.  I've added this to that change and I'll submit that once this is merged.  That way I can keep those changes together.